lib/vision/index.js

```diff
@@ -0,0 +1,773 @@
+
+/*!
+ * @module vision
+ */
+
+'use strict';
+
+var arrify = require('arrify');
+var async = require('async');
+var extend = require('extend');
+var format = require('string-format-obj');
+var fs = require('fs');
+var is = require('is');
+var nodeutil = require('util');
+var prop = require('propprop');
+
+/**
+ * @type {module:storage/file}
+ * @private
+ */
+var File = require('../storage/file.js');
+
+/**
+ * @type {module:common/service}
+ * @private
+ */
+var Service = require('../common/service.js');
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+var VERY_UNLIKELY = 0;
+var UNLIKELY = 1;
+var POSSIBLE = 2;
+var LIKELY = 3;
+var VERY_LIKELY = 4;
+
+/**
+ * The [Cloud Vision API](https://cloud.google.com/vision/docs) allows easy
+ * integration of vision detection features, including image labeling, face and
+ * landmark detection, optical character recognition (OCR), and tagging of
+ * explicit content.
+ *
+ * @constructor
+ * @alias module:vision
+ *
+ * @classdesc
+ * <p class="notice">
+ *   **This is a Beta release of Google Cloud Vision.** This API is not covered
+ *   by any SLA or deprecation policy and may be subject to backward-
+ *   incompatible changes.
+ * </p>
+ *
+ * To learn more about the Vision API, see
+ * [Getting Started](https://cloud.google.com/vision/docs/getting-started).
+ *
+ * @resource [Getting Started]{@link https://cloud.google.com/vision/docs/getting-started}
+ *
+ * @param {object} options - [Configuration object](#/docs).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var translate = gcloud.vision();
+ */
+function Vision(options) {
+  if (!(this instanceof Vision)) {
+    options = util.normalizeArguments(this, options);
+    return new Vision(options);
+  }
+
+  var config = {
+    baseUrl: 'https://vision.googleapis.com/v1',
+    projectIdRequired: false,
+    scopes: [
+      'https://www.googleapis.com/auth/cloud-platform'
+    ]
+  };
+
+  Service.call(this, config, options);
+}
+
+nodeutil.inherits(Vision, Service);
+
+Vision.likelihood = {
+  VERY_UNLIKELY: VERY_UNLIKELY,
+  UNLIKELY: UNLIKELY,
+  POSSIBLE: POSSIBLE,
+  LIKELY: LIKELY,
+  VERY_LIKELY: VERY_LIKELY
+};
+
+/**
+ * Run image detection and annotation for an image or batch of images.
+ *
+ * @resource [images.annotate API Reference]{@link https://cloud.google.com/vision/reference/rest/v1/images/annotate}
+ *
+ * @param {object|object[]} requests - An `AnnotateImageRequest` or array of
+ *     `AnnotateImageRequest`s. See an
+ *     [`AnnotateImageRequest`](https://cloud.google.com/vision/reference/rest/v1/images/annotate#AnnotateImageRequest).
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * vision.annotate();
```

Comment:
I realize we probably don't want to try and give an example of all the available options here, but considering the request and callback params are not optional, I think at the very least we should include them, even if they're empty or noop.